pywbem/cim_operations.py

```diff
@@ -1709,6 +1721,269 @@ def OpenEnumerateInstancePaths(self, ClassName, namespace=None,
                 self.operation_recorder.record_staged()
             return result_tuple
 
+    # TODO overriding the internal choice of pull vs enum. Should we be able
+    #      force one or the other for a connection?
```

Comment:
On overriding: No. The user can already control the behavior of the generator method using the `has_pulled_operations` capability of the connection. If the user wants a different behavior on that same connection than the behavior selected with `has_pulled_operations`, they can use the specific pulled or non-pulled operation methods directly. First, there is no performance disadvantage of doing that compared to using the generator method, because the generator method still materializes the result set of each pull or (in case of non-pulled) of the entire operation, before it yields its instances. Admittedly, a little more coding is needed for using the pull operations directly. Second, cases where it makes sense to use the opposite of what was selected when creating the connection should be rare.